ndk/src/events/encryption.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * Encryption and gift-wrapping of events
+ * Implements Nip04, Nip44, Nip59 
+ */
+import type { NDKSigner } from "../signers";
+import { NDKUser } from "../user";
+import { NDKEvent } from "./index.js";
+
+// NIP04 && NIP44
+
+export type EncryptionNip = 'nip04' | 'nip44';
+export type EncryptionMethod = 'encrypt' | 'decrypt'
+
+// some clients may wish to set a default for message encryption...
+let defaultEncryption : EncryptionNip | undefined = undefined;
+export function useEncryption(nip : EncryptionNip) {
+    defaultEncryption = nip;
+}
+
+export async function encrypt(
+    this: NDKEvent,
+    recipient?: NDKUser,
+    signer?: NDKSigner,
+    nip : EncryptionNip | undefined = defaultEncryption
+): Promise<void> {
+    let encrypted : string | undefined;
+    if (!this.ndk) throw new Error("No NDK instance found!");
+    if (!signer) {
+        this.ndk.assertSigner();
+        signer = this.ndk.signer;
+    }
+    if(!signer) throw new Error('no NDK signer');
```

Comment:
As far as I remember, `assertSigner()` will throw an error if there isn't a signer available.

Reply:
Good catch, I'll remove the extra check.